metric-schema-java/src/main/java/com/palantir/metric/schema/Validator.java

```diff
@@ -24,7 +24,7 @@
 final class Validator {
 
     private static final String SHORT_NAME_PATTERN = "[A-Z][a-zA-Z0-9]+";
-    private static final String NAME_SEGMENT_PATTERN = "[a-z][a-zA-Z0-9\\-]+";
+    private static final String NAME_SEGMENT_PATTERN = "[a-z0-9][a-zA-Z0-9\\-]*";
     private static final String NAME_PATTERN = "(" + NAME_SEGMENT_PATTERN + "\\.)*" + NAME_SEGMENT_PATTERN;
```

Comment:
curious, do we want to only allow single digit or single character segment only for the last segment?

Reply:
I think we were overzealous with the initial validation, WC has a few metrics based on status code, I don't think we need to be too prescriptive.